src/mpid/ch4/src/ch4_part.h

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#ifndef CH4_PART_H_INCLUDED
+#define CH4_PART_H_INCLUDED
+
+#include "ch4_impl.h"
+
+MPL_STATIC_INLINE_PREFIX int MPIDI_part_start(MPIR_Request * request)
+{
+    int mpi_errno = MPI_SUCCESS;
+
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_PART_STARTALL);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_PART_STARTALL);
+
+    MPID_THREAD_CS_ENTER(VCI, MPIDI_VCI(0).lock);
+    MPIDI_workq_vci_progress_unsafe();
+
+#ifdef MPIDI_CH4_DIRECT_NETMOD
+    mpi_errno = MPIDI_NM_part_start(request);
+#else
+    if (MPIDI_REQUEST(request, is_local)) {
+        mpi_errno = MPIDI_SHM_part_start(request);
+    } else {
+        mpi_errno = MPIDI_NM_part_start(request);
+    }
+#endif
+    MPIR_ERR_CHECK(mpi_errno);
+
+    MPIDI_PART_REQUEST(request, active_flag) = 1;
+
+  fn_exit:
+    MPID_THREAD_CS_EXIT(VCI, MPIDI_VCI(0).lock);
+    MPIR_FUNC_VERBOSE_EXIT(MPID_STATE_MPIDI_PART_STARTALL);
+    return mpi_errno;
+  fn_fail:
+    goto fn_exit;
+}
+
+MPL_STATIC_INLINE_PREFIX int MPID_Pready_range(int low, int high, MPIR_Request * request)
```

Comment:
The parameter names here are different from the netmod API (`low` vs `partition_low`). I would prefer shorter names and I know the longer names are from MPI spec. But either way, I think we should keep them consistent unless with specific reasons.

Reply:
Will change to be consistent with the spec names.